CMakeLists.txt

```diff
@@ -1044,7 +1044,9 @@ if(WIN32 AND (NOT MINGW))
 
 elseif(UNIX)
 
-  if (CMAKE_CROSSCOMPILING AND (NOT ENABLE_HOST_CPU_DEVICES) AND (NOT ENABLE_HSA))
+  # proxy on android uses opencl-stub that requires dlfcn.
+  if (CMAKE_CROSSCOMPILING AND (NOT ENABLE_HOST_CPU_DEVICES) AND (NOT ENABLE_HSA)
+        AND (NOT (ENABLE_PROXY_DEVICE AND ANDROID)))
```

Comment:
The proxy only dlopens when `PROXY_USE_LIBOPENCL_STUB` is set and not limited to just Android, so I'd check for that instead.